proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+class PermissionFilter {
+    PermissionFilter(UUID identifier, List<PermissionElement> permissionElements, int groupType, PermissionManager.OnPermissionChangeListener listener) {}
+
+    UUID getIdentifier() {}
+
+    List<PermissionElement> getPermissionElements() {}
+
+    int getGroupType() {}
+
+    PermissionManager.OnPermissionChangeListener getListener() {}
+}
+```
+
+### Other Additions
+
+#### SDLProxyBridge
+
+The SDLProxyBridge has been implemented to allow the library to bridge from old API's to new. The `SdlManager` will hold an instantiate of this class that is passed to its proxy object. Using the `SdlProxyBase` as our main interface into the old APIs with this bridge the manager API change can be added with only a minor version change. Proxy bridge allows the managers to listen for specific RPCs without having to individually implement `IProxyListenerBase` themselves and having to override all of the response methods like what is currently required.
+
+```java
+public class ProxyBridge implements IProxyListenerBase{
+
+	public interface OnRPCListener {
+		void onRpcReceived(int functionID, RPCMessage message);
+	}
+
+	protected interface LifecycleListener{
+		void onProxyClosed(String info, Exception e, SdlDisconnectedReason reason);
+		void onServiceEnded(OnServiceEnded serviceEnded);
+		void onServiceNACKed(OnServiceNACKed serviceNACKed);
+		void onError(String info, Exception e);
+	}
+
+	public ProxyBridge( LifecycleListener lifecycleListener){}
+
+	public boolean onRPCReceived(final RPCMessage message){}
+
+	protected void addRpcListener(FunctionID id, OnRPCListener listener){}
+
+	public boolean removeOnRPCListener(FunctionID id, OnRPCListener listener){}
+	
+	.....
+	
+	// All overridden IProxyListenerBase methods
+```
+
+#### SDLFile
+
+SDLFile will be added to further match the API with iOS. It will contain helpers that do some of the work that we required the developer to do before, like automatically setting the file type and assembling the byte array of the file to be sent. For uploading a file, the developer may choose to pass either a resource id, a URI, or the byte array itself. 
+
+
+```java
+public SdlFile{
+
+	public void setName(@NonNull String fileName) 
+	public String getName()
+
+	public void setResourceId(int id, Resources resources)
+	public void getResourceId()
+
+	public void setUri(URI uri)
+	public byte[] getUri()
+
+	public void setFileData(byte[] data)
+	public byte[] getFileData()
+
+	public void setType(@NonNull FileType fileType)
+	public FileType getType()
+
+	public void setPersistent(boolean persistentFile)
+	public boolean isPersistent()
+}
+```
+
+###### SDLArtwork
+
+`SDLArtwork` will extend `SDLFile` and be specifically for uploading images.
+
+
+#### CompletionListener
+
+A simple listener with a single method, `onComplete(boolean success)` will be added. It's purpose is simply understand when an asynchronous call int he library was completed, and if it was completed successfully or not. 
```

Comment:
Please update second sentence to "Its purpose is simply to understand when an asynchronous call in the library..."